python/dgl/nn/pytorch/glob.py

```diff
@@ -0,0 +1,673 @@
+                               dropouth=dropouth, dropouta=dropouta) for _ in range(2)])
+        self._reset_parameters()
+
+    def _reset_parameters(self):
+        nn.init.xavier_uniform_(self.I)
+
+    def forward(self, feat, graph, isab_graph=None):
+        """
+        Compute an Induced Set Attention Block.
+
+        Parameters
+        ----------
+        feat : torch.Tensor
+            The input feature
+        graph : DGLGraph
+            The graph.
+        isab_graph: DGLGraph or None
+            The graph to apply Message Passing on, set to None if not
+            specified.
+        """
+        query = self.I
+        batch_size = 1
+        if isinstance(graph, BatchedDGLGraph):
+            batch_size = graph.batch_size
+
+        if isab_graph is None:
+            induced_graph = _create_graph_from_num_nodes([self.m] * batch_size)
+            isab_graph = [
+                _create_bipartite_graph(graph, induced_graph),
+                _create_bipartite_graph(induced_graph, graph)
+            ]
+
+        query = query.repeat(batch_size, 1)
+        for mha, (g, kv_nids, q_nids) in zip(self.mha, isab_graph):
+            rst = mha(g, query, feat, q_nids, kv_nids)
+            query, feat = feat, rst
+
+        return rst
+
+    def extra_repr(self):
+        """Set the extra representation of the module.
+        which will come into effect when printing the model.
+        """
+        shape_str = '({}, {})'.format(self.I.shape[0], self.I.shape[1])
+        return 'InducedVector: ' + shape_str
+
+
+class PMALayer(nn.Module):
+    r"""Pooling by Multihead Attention, used in the Decoder Module of Set Transformer."""
+    def __init__(self, k, d_model, num_heads, d_head, d_ff, dropouth=0., dropouta=0.):
+        self.k = k
+        super(PMALayer, self).__init__()
+        self.S = nn.Parameter(
+            th.FloatTensor(k, d_model)
+        )
+        self.mha = MultiHeadAttention(d_model, num_heads, d_head, d_ff,
+                                      dropouth=dropouth, dropouta=dropouta)
+        self._reset_parameters()
+
+    def _reset_parameters(self):
+        nn.init.xavier_uniform_(self.S)
+
+    def forward(self, feat, graph, pma_graph=None):
+        """
+        Compute Pooling by Multihead Attention.
+
+        Parameters
+        ----------
+        feat : torch.Tensor
+            The input feature
+        graph : DGLGraph
+            The graph.
+        pma_graph: DGLGraph or None
+            The graph to apply Message Passing on, set to None if not
+            specified.
+
+        Returns
+        -------
+        torch.Tensor
+            The output feature
+        """
+        query = self.S
+        batch_size = 1
+        if isinstance(graph, BatchedDGLGraph):
+            batch_size = graph.batch_size
+
+        if pma_graph is None:
+            induced_graph = _create_graph_from_num_nodes([self.k] * batch_size)
+            pma_graph = _create_bipartite_graph(graph, induced_graph)
```

Comment:
I wonder what's the relation between this bipartite graph and the to-be implemented bipartite graph in heterograph context. Will you reuse that? Also, I feel these graph construction routines are quite common. Maybe consider expose them in dgl main namespace for end users.